### DISCUSSION TITLE
Technology/Tool: Biome

### DISCUSSION THREAD

**Discussion by alexanderniebuhr** [astrolicious/astrolicious#8] [Board]
## Background:

The efficiency of maintainers is significantly enhanced by having consistent formatting and listing rules across all repositories. To achieve this goal, the necessary tooling should be minimally intrusive, allowing developers to focus on their work without unnecessary interruptions. It is crucial that such tools are performant, reliable, and opinionated, ensuring they operate seamlessly within our Continuous Integration and Continuous Deployment (CI/CD) pipelines.

## Proposal

We propose the adoption of Biome for formatting and linting tasks within two main project types: organizational projects and onboarded community projects. We recognize the autonomy of standalone projects and, as such, do not require them to adopt this new tooling framework.

While ESLint and Prettier are commendable tools widely embraced for their utility, they do not align with our long-term vision. In contrast, Biome offers a unified solution for both formatting and linting that is designed to integrate effortlessly with CI/CD pipelines.

To facilitate this transition, we plan to provide comprehensive documentation and guidelines on Biome's usage. Additionally, we will integrate Biome into all project templates provided by Astrolicious.

## Pros
- High performance, ensuring minimal disruption to development workflows.
- Comparability with Prettier in terms of formatting quality.
- A single tool solution for both formatting and linting needs.
- Seamless integration with CI/CD pipelines, promoting efficiency and reliability.

## Cons
- Currently lacks extensive formatting capabilities, though this is addressed in the roadmap for future updates.
- Absence of support for typed linting, which may necessitate supplementary tools in the interim.

## Conclusion

The advantages of adopting Biome significantly outweigh the disadvantages. Its performance and integration capabilities make it a valuable tool for our projects. We are confident in Biome's ongoing development and its ability to meet our needs progressively. Nevertheless, we remain open to reevaluating our tooling choices to ensure they consistently meet our standards of efficiency and effectiveness.

**Comment by florian-lefebvre** [comment]
I'm in favor of this! The less tools the better! The only pain point right now is template formatting that is not supported yet
